### PR TITLE
Add support for sanitizing HTTP header values.

### LINKED ISSUE #1184: Add more features for adding HTTP request / response headers to spans.
I already have https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1172 open for this, and I'll be breaking it in to smaller pieces at @lzchen 's request.

**Is your feature request related to a problem?**
Currently, you can only provide a list of full HTTP request / response header names to be added to the span.

There is also no capacity for header value redaction.

**Describe the solution you'd like**
It would be nice to be able to specify a regex or "all" to get all headers.

Header value redaction is also a must-have for us.

**Describe alternatives you've considered**
I considered doing this in my application, but it makes more sense to add it here.

### PR DESCRIPTION
**Pull request by rogersd**
# Description

First step of https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1184

This provides basic support to allow for sanitizing of HTTP headers.  Future commits will add code to the various instrumentations that use this functionality.

Fixes #1184

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Unit test added.

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [X] Documentation has been updated

### CODE REVIEW

**Comment by rogersd** [comment]
@srikanthccv @lzchen Could you please take a look at this so I can continue opening PRs?

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +43,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1197](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1197))
 - Add metric instumentation for flask
   ([#1186](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1186))
+- `opentelemetry-util-http` Add support for sanitizing HTTP header values.
```
Comment (srikanthccv) [inline CHANGELOG.md:46]: Please move this to unreleased section

File: util/opentelemetry-util-http/tests/test_capture_custom_headers.py [hunk 2]
```diff
@@ -58,6 +59,21 @@ def test_get_custom_response_header(self):
...
+    )
+    def test_get_custom_sanitize_header(self):
+        custom_headers_to_capture = get_custom_headers(
+            OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS
+        )
+        self.assertEqual(
+            custom_headers_to_capture,
+            ["My-Secret-Header", "My-Secret-Header-2"],
+        )
+
```
Comment (srikanthccv) [inline util/opentelemetry-util-http/tests/test_capture_custom_headers.py:76]: Add test for `sanitize_header_value`

File: CHANGELOG.md (hunk 1 above)
Comment (rogersd) [reply CHANGELOG.md]: Yeah, this keeps ending up in the wrong place after `main` gets merged in to my branch.

File: util/opentelemetry-util-http/tests/test_capture_custom_headers.py (hunk 2 above)
Comment (rogersd) [reply util/opentelemetry-util-http/tests/test_capture_custom_headers.py:77]: Done.

File: CHANGELOG.md (hunk 1 above)
Comment (rogersd) [reply CHANGELOG.md]: Fixed.

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**